### PR TITLE
chore: refactor and fix resend OTP

### PR DESCRIPTION
**Pull request by EugeneDevv**

### CODE REVIEW

**Review by byronkimani — APPROVED** [review]

File: lib/application/redux/actions/resend_otp_action.dart [hunk 1]
```diff
@@ -3,59 +3,53 @@ import 'dart:convert';
...
-  final BuildContext context;
   final String userName;
+  final String reSendOTPEndpoint;
+  final IGraphQlClient httpClient;
+  final void Function(String reason)? errorCallback;
 
   @override
   void after() {
-    dispatch(WaitAction<AppState>.add(resendOTPFlag));
+    dispatch(WaitAction<AppState>.remove(sendOTPFlag));
```
Comment (abiudrn) [inline lib/application/redux/actions/resend_otp_action.dart:33]: This flag `sendOTPFlag` is being used in a `ResendOTPAction`. You should consider changing one... either the action name or the flag name

File: lib/presentation/onboarding/verify_phone/widgets/verify_otp_widget.dart [hunk 2]
```diff
@@ -122,25 +121,20 @@ class VerifyOtpWidgetState extends State<VerifyOtpWidget>
...
-                    restartTimer();
-                    textEditingController.clear();
-                    await Future<dynamic>.delayed(const Duration(seconds: 3));
-                  },
-                )
-              else ...<Widget>[
-                smallVerticalSizedBox,
-                widget.loader,
-              ]
+              MyAfyaHubPrimaryButton(
```
Comment (abiudrn) [inline lib/presentation/onboarding/verify_phone/widgets/verify_otp_widget.dart:124]: Why did you remove the `if (!widget.vm.wait!.isWaitingFor(resendOTPFlag))`?

It was very vital initially to prevent the user from sending multiple OTPs even when an error occurs...

**Review by abiudrn — CHANGES_REQUESTED** [review]

**Review by abiudrn — APPROVED** [review]
